drivers/i2c/i2c_ite_enhance.c

```diff
@@ -318,11 +319,21 @@ static void i2c_enhanced_port_set_frequency(const struct device *dev,
 		}
 
 		/* Adjust SCL low period prescale */
-		psr += config->prescale_scl_low;
+		psr_l = psr + config->prescale_scl_low;
+		if (psr_l > 0xFD) {
+			psr_l = 0xFD;
+		}
+
+		/* Adjust SCL high period prescale */
+		psr_h = psr - config->prescale_scl_low;
```

Comment:
This calculation could underflow.  You should verify that `config->prescale_scl_low` is less than `psr`.

Reply:
Done.